crypto/entropy.go

```diff
@@ -0,0 +1,36 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+)
+
+func GenerateEntropy(n int) ([]byte, error) {
+	if n <= 0 {
+		return nil, errors.New("entropy byte size must be > 0")
+	}
+
+	bytes := make([]byte, n)
+	size, err := rand.Read(bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	if size != n {
```

Comment:
Unclear to me why `rand.Read()` even returns the `n int` (I called it `size`??) when the docs https://pkg.go.dev/crypto/rand#Read state `On return, n == len(b) if and only if err == nil` ... in other words they're semantically equivalent. Under what circumstance would `err` not be `nil` but `n` wouldn't be equal to the `len(b)`? 

In other rods, do we even need this condition?

Reply:
I'm making a guess based of reading the comments in the [go source code](https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/io/io.go;l=346-352). Returning `n` seems to be more useful in adding detail to the error case rather than informing the success case. If the byte slice is partially filled, `rand.Read` will return both the number of bytes filled AND the error.

Since the (rather minimal) docs for `rand.Read` say that `On return, n == len(b) if and only if err == nil`, it's safe to omit this `if size != n` check since we have already checked `if error != nil`.

Reply:
Love it! Removed!

Reply:
For future reference the semantics are fully documented in the [io.Reader](https://pkg.go.dev/io#Reader) interface. All the core IO abstractions are in this package, and other packages just implement them.